### PR TITLE
[ui] Improve wrapping on asset catalog page

### PR DESCRIPTION
**Pull request by clairelin135**
Improve wrapping for smaller screen sizes on the asset catalog page. Linear ticket: https://linear.app/dagster-labs/issue/FE-244/handle-long-strings-in-asset-catalog-page

Adds:
- Ellipsis when text overflows
- When screen size is below 800px, switch to rows instead of 3-column layout

image
image

### CODE REVIEW

**Comment by clairelin135** [comment]
* **#20457** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20457?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clairelin135 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by salazarm — CHANGES_REQUESTED** [review]
Switch to display: flex or display: grid

**Comment by clairelin135** [comment]
@salazarm thanks, didn't realize `display:grid` was a thing. I think this is looking better now, would appreciate a second look!

**Review by salazarm — APPROVED** [review]
A lot cleaner, thanks for fixing!

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx [hunk 1]
```diff
@@ -403,7 +405,24 @@ const SectionName = styled.span`
...
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+};
+
+const GridStyle = styled.div`
+  padding: 16px 12px;
+
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr;
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:418]: I think you want `minmax(0, 1fr)`

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx [hunk 2]
```diff
@@ -306,85 +300,93 @@ export const AssetsOverview = ({viewerName}: {viewerName?: string}) => {
...
                     key={repoAddressAsHumanString(countPerCodeLocation.repoAddress)}
                     assetsCount={countPerCodeLocation.assetCount}
                   >
                     <Icon name="folder" />
-                    <Link to={linkToCodeLocation(countPerCodeLocation.repoAddress)}>
+                    <Link
+                      to={linkToCodeLocation(countPerCodeLocation.repoAddress)}
+                      style={TextOverflowStyle}
+                    >
                       {repoAddressAsHumanString(countPerCodeLocation.repoAddress)}
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:385]: I belie

Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:385]: I believe you want to <MiddleTruncate> this text

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx [hunk 3]
```diff
@@ -403,7 +405,24 @@ const SectionName = styled.span`
...
+  textOverflow: 'ellipsis',
+};
+
+const GridStyle = styled.div`
+  padding: 16px 12px;
+
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr;
+  @media (max-width: 768px) {
+    grid-template-columns: 1fr;
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:420]: same

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx [hunk 4]
```diff
@@ -403,7 +405,24 @@ const SectionName = styled.span`
...
+  }
+`;
+
+const GridItem = styled.div`
+  padding: 6px 12px;
+  display: grid;
+  grid-template-columns: auto auto;
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:427]: you don't need display grid here if you're just going to `auto auto`;
display flex is fine.

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:427]: (unless you have an actual grid -- more than 1 row, which I think you dont here?)

Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:427]: Kind of a marginal difference, but with this grid we see:
image


And with flex instead, we see:
image

Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:427]: The grid version seems to display a bit more cleanly but the flexbox version displays more content?

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:427]: oh I think in this case we want the left side to be able to shrink if we go with flex

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx (hunk 2 above)
Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:388]: ahh good call, changed

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx (hunk 4 above)
Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx]: It does truncate.. not sure if I need to add other CSS properties. I switched this to use flexbox though

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx [hunk 5]
```diff
@@ -415,14 +418,14 @@ const GridStyle = styled.div`
   padding: 16px 12px;
 
   display: grid;
-  grid-template-columns: 1fr 1fr 1fr;
+  grid-template-columns: minmax(0, 1fr) minmax(0, 1fr) minmax(0, 1fr);
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx:421]: oh forgot to tell you that you can do `repeat(3, minmax(0, 1fr))`

**Pull request merged**